app/main.c

```diff
@@ -315,16 +313,26 @@ int app_main(bool is_resale)
 
 #endif
 	if (is_ownership_transfer(do_resale)) {
-		return 0;
+		ret = 0;
+		goto end;
 	}
 
 	print_device_status();
 
 	if (FDO_SUCCESS != fdo_sdk_run()) {
 		LOG(LOG_ERROR, "FIDO Device Onboard failed\n");
-		return -1;
+		ret = -1;
+		goto end;
 	}
 
+	ret = 0;
+end:
+	/* free the module related info
+	 * FDO has created the required DB
+	 */
+	free(module_info);
```

Comment:
This would be dereferenced by 'goto end' at 'configure_blobs()'. Please add if (module_info) check before freeing.